### PR TITLE
Add encoding to FossId credentials in URL mappings

### PR DESCRIPTION
**Pull request by oheger-bosch**
Refer to the single commits for further details.

### CODE REVIEW

File: scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt [hunk 1]
```diff
@@ -66,7 +66,15 @@ internal class FossIdUrlProvider private constructor(
          */
         fun create(
             urlMapping: Collection<String> = emptyList()
-        ): FossIdUrlProvider = FossIdUrlProvider(urlMapping.toRegexMapping())
+        ): FossIdUrlProvider {
+            val mappings = urlMapping.toRegexMapping()
+            logger.info { "${mappings.size} mappings have been loaded." }
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt:71]: Do we really need this if we have the following? I.e. do we expect there to be so many mappings that it's too hard to count them manually (eventually by looking at log line numbers)?

File: scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt [hunk 2]
```diff
@@ -144,9 +152,12 @@ internal class FossIdUrlProvider private constructor(
             if (replaced != repoUrl) {
                 logger.info { "URL mapping applied to $repoUrl: Mapped to $replaced." }
                 return replaceVariables(replaced)
+            } else {
+                logger.debug { "Mapping $regex did not match." }
             }
         }
 
+        logger.info { "No matching URL mapping could be found for '$repoUrl'." }
```
Comment (sschuberth) [inline scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt:160]: Nit: Empty line after this one to separate from the unrelated `return`  statement.

File: scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt [hunk 3]
```diff
@@ -66,7 +66,15 @@ internal class FossIdUrlProvider private constructor(
          */
         fun create(
             urlMapping: Collection<String> = emptyList()
-        ): FossIdUrlProvider = FossIdUrlProvider(urlMapping.toRegexMapping())
+        ): FossIdUrlProvider {
+            val mappings = urlMapping.toRegexMapping()
+            logger.info { "${mappings.size} mappings have been loaded." }
+            mappings.forEach {
+                logger.info { "Mapping ${it.key} -> ${it.value}." }
```
[the remark this answers is not in the document]
Comment (oheger-bosch) [reply scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt:73]: Maybe as a compromise: We remove the line with the size, but leave the level here at info?

[the remark this answers is not in the document]
Comment (sschuberth) [reply scanner/src/main/kotlin/scanners/fossid/FossIdUrlProvider.kt:72]: Ok with me.

**Review by sschuberth — APPROVED** [review]

**Pull request merged**